### ISSUE TITLE
Matches starting at the start of files are not properly handled

### ISSUE THREAD

**Issue by hsahovic** [dalance/flexlint#156]
# Description

There's an inconsistency in the behavior of this tool: when a file starts with the forbidden pattern, the tool recognizes the error but doesn't report it properly and doesn't exit with a non-zero exit code.

# Example setup

## Degenerate case

**`.flexlint.toml`:**

```toml
[[rules]]
name        = "Example c test"
pattern     = 'hello'
forbidden   = 'hello'
hint        = "Hello is forbidden"
includes    = ["**/*.c"]
```

**`a.c`:**

```c
hello
```

Output:

```
$ flexlint
- Summary ----------------------------------------------------------------------

  * Checked files : 5
  * Checked points: 1 ( Pass: 0, Fail: 1, Skip: 0 )
  * Elapsed time  : 0.00020999999999999998s
```

Notice that the report does mention 1 fail, but does not report it. Additionally, the exit code is 0 - this wouldn't fail in a CI pipeline.

## Normal case

If we add a newline at the start of the c file, eg.

```c

hello
```

The output becomes:

```
$ flexlint
Fail: Example c test
   --> a.c:2:1
  |
2 | hello
  | ^^^^^ hint: Hello is forbidden

- Summary ----------------------------------------------------------------------

  * Checked files : 5
  * Checked points: 1 ( Pass: 0, Fail: 1, Skip: 0 )
  * Elapsed time  : 0.000244s
```

With a non-0 exit code. We expect a similar output in the first case, which we do not get.

# Cause

For the matches not being displayed, this is cause by `pos` being initialized at `0` and incremented before checking any match, at this line: https://github.com/dalance/flexlint/blob/ffa8e0ef632fd762157872d5cfcfd9625c9e5cb5/src/printer.rs#L174 
For the non-0 exit code, this is caused by `all_pass` being defined in `printer.rs` file. It's initialized as `true`, and is set to `false` when an error is displayed: https://github.com/dalance/flexlint/blob/ffa8e0ef632fd762157872d5cfcfd9625c9e5cb5/src/printer.rs#L202

This can be fixed by incrementing `pos` after checking all matches, rather than before.

This is implemented in https://github.com/dalance/flexlint/pull/155, along minor changes to the actual printing to avoid panics in both normal and simple modes.

**Comment by hsahovic** [comment]
@dalance with the PR meged, I think this can be closed, unless you want to keep it up for now?
